### PR TITLE
Dev datamonitor v1

### PR DESCRIPTION
**Pull request by IonutMuthi**
Data logger version 2.0 first draft

### CODE REVIEW

File: plugins/datamonitor/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+target_include_directories(${PROJECT_NAME} PUBLIC scopy-pluginbase)
+
+foreach(comp ${SCOPY_QT_COMPONENTS})
+	set(SCOPY_QT_LIBRARIES ${SCOPY_QT_LIBRARIES} Qt${QT_VERSION_MAJOR}::${comp})
+endforeach()
+
+target_link_libraries(
+	${PROJECT_NAME}
+	PUBLIC Qt5::Widgets
+	       libm2k::libm2k
```
Comment (adisuciu) [inline plugins/datamonitor/CMakeLists.txt:77]: There is no dependency to libm2k in this plugin

File: plugins/datamonitor/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+	       scopy-pluginbase
+	       scopy-iioutil
+	       scopy-gui
+	       scopy-core
+	       ${QWT_LIBRARIES}
+	       ${GLIB_LINK_LIBRARIES}
```
Comment (adisuciu) [inline plugins/datamonitor/CMakeLists.txt:83]: There should be no dependency to GLIB in thisn plugin ..

File: plugins/datamonitor/test/tst_pluginloader.cpp [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+	void qobjectcast_to_plugin();
+	void clone();
+	void name();
+	void metadata();
+};
+
+#define PLUGIN_LOCATION "../../plugins"
+#define FILENAME PLUGIN_LOCATION "/libscopy-bareminimumplugin.so"
+
+void TST_BareMinimum::fileExists()
```
Comment (adisuciu) [inline plugins/datamonitor/test/tst_pluginloader.cpp:29]: This test is outdated. It should use your own plugin .so and should be named accordingly.

**Review by adisuciu — CHANGES_REQUESTED** [review]

File: plugins/datamonitor/include/datamonitor/datamonitor.h [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+public Q_SLOTS:
+	bool onConnect() override;
+	bool onDisconnect() override;
+
+private:
+	DataMonitorTool *tool;
+	IIOPingTask *ping;
+	CyclicalTask *cs;
+
+	libm2k::context::Context *libm2k_context;
```
Comment (AlexandraTrifan) [inline plugins/datamonitor/include/datamonitor/datamonitor.h:60]: are we using the libm2k context anywhere?

File: plugins/datamonitor/src/datamonitor.cpp [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+#include <iio.h>
+
+#include <QDebug>
+#include <QLabel>
+#include <QUuid>
+#include <QVBoxLayout>
+#include <menucollapsesection.h>
+#include <menusectionwidget.h>
+
+#include <libm2k/contextbuilder.hpp>
```
Comment (AlexandraTrifan) [inline plugins/datamonitor/src/datamonitor.cpp:14]: same as previous comments, libm2k is not used here

File: plugins/datamonitor/src/datamonitortool.cpp [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+
+	tool->setRightContainerWidth(300);
+	tool->centralContainer()->setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Preferred);
+
+	tool->openLeftContainerHelper(false);
+	tool->openRightContainerHelper(false);
+
+	lay->addWidget(tool);
+
+	openLatMenuBtn = new OpenLastMenuBtn(dynamic_cast<MenuHAnim *>(tool->rightContainer()), false, this);
```
Comment (AlexandraTrifan) [inline plugins/datamonitor/src/datamonitortool.cpp:50]: typo? shoule be openLastMenuBtn across the plugin

File: plugins/datamonitor/src/datamonitor.cpp
Comment (AlexandraTrifan) [inline plugins/datamonitor/src/datamonitor.cpp:1]: shouldn't this file be called datamonitorplugin.cpp for consistency across plugins?

File: plugins/CMakeLists.txt
Comment (AlexandraTrifan) [inline plugins/CMakeLists.txt:1]: It seems like you're missing some destructors and need to do some object cleanup.

**Review by adisuciu — APPROVED** [review]

**Pull request merged**